app/backend/app.py

```diff
@@ -205,15 +229,69 @@ def config():
             "showGPT4VOptions": current_app.config[CONFIG_GPT4V_DEPLOYED],
             "showSemanticRankerOption": current_app.config[CONFIG_SEMANTIC_RANKER_DEPLOYED],
             "showVectorOption": current_app.config[CONFIG_VECTOR_SEARCH_ENABLED],
+            "showUserUpload": current_app.config[CONFIG_USER_UPLOAD_ENABLED],
         }
     )
 
 
+@bp.post("/upload")
+@authenticated
+async def upload(auth_claims: dict[str, Any]):
+    request_files = await request.files
+    if "file" not in request_files:
+        # If no files were included in the request, return an error response
+        return jsonify({"message": "No file part in the request", "status": "failed"}), 400
+
+    user_oid = auth_claims["oid"]
+    file = request_files.getlist("file")[0]
+    user_blob_container_client: FileSystemClient = current_app.config[CONFIG_USER_BLOB_CONTAINER_CLIENT]
+    user_directory_client = user_blob_container_client.get_directory_client(user_oid)
+    await user_directory_client.set_access_control(owner=user_oid)
+    file_client = user_directory_client.get_file_client(file.filename)
+    file_io = file
+    file_io.name = file.filename
+    file_io = io.BufferedReader(file_io)
+    await file_client.upload_data(file_io, overwrite=True, metadata={"UploadedBy": user_oid})
+    file_io.seek(0)
+    ingester = current_app.config[CONFIG_INGESTER]
+    await ingester.add_file(File(content=file_io, acls={"oids": [user_oid]}))
+    return jsonify({"message": "File uploaded successfully"}), 200
+
+
+@bp.post("/delete_uploaded")
+@authenticated
+async def delete_uploaded(auth_claims: dict[str, Any]):
+    request_json = await request.get_json()
+    filename = request_json.get("filename")
+    user_oid = auth_claims["oid"]
+    user_blob_container_client: FileSystemClient = current_app.config[CONFIG_USER_BLOB_CONTAINER_CLIENT]
+    user_directory_client = user_blob_container_client.get_directory_client(user_oid)
+    file_client = user_directory_client.get_file_client(filename)
+    await file_client.delete_file()
+    ingester = current_app.config[CONFIG_INGESTER]
+    await ingester.remove_file(filename, user_oid)
+    return jsonify({"message": f"File {filename} deleted successfully"}), 200
+
+
+@bp.get("/list_uploaded")
+@authenticated
+async def list_uploaded(auth_claims: dict[str, Any]):
+    user_oid = auth_claims["oid"]
+    user_blob_container_client: FileSystemClient = current_app.config[CONFIG_USER_BLOB_CONTAINER_CLIENT]
+    all_paths = user_blob_container_client.get_paths(path=user_oid)
```

Comment:
Annoying, you can't just get directory_client and call get_paths on it, this seems to be the only way.

Reply:
correct, the reason why is that this call is paging potentially